module/VuFind/src/VuFind/Search/SearchOrigin/AbstractSearchOrigin.php

```diff
@@ -67,4 +67,6 @@ abstract public function getSearchUrlParamsArray(): array;
      * @return array
      */
     abstract public function getOriginUrlParamsArray(): array;
+
+    abstract public function getUrlName(): string;
```

Comment:
I would use "route name" rather than "URL name" terminology for consistency with the rest of the code.